### ISSUE TITLE
invalidateInput

### ISSUE THREAD

**Issue by fcorneli** [omnifaces/omnifaces#748]
Within our JSF projects we start having the following utility methods all over the place:
```java
public static void invalidateInput(String... relativeIds) {
        FacesContext facesContext = FacesContext.getCurrentInstance();
        UIComponent currentComponent = UIComponent.getCurrentComponent(facesContext);
        if (null == currentComponent) {
            LOGGER.error("no current component");
            return;
        }
        for (String relativeId : relativeIds) {
            UIComponent component = currentComponent.findComponent(relativeId);
            if (null == component) {
                LOGGER.error("component not found: {}", relativeId);
                continue;
            }
            if (!(component instanceof UIInput)) {
                LOGGER.error("component is not a UIInput: {}", component.getId());
                continue;
            }
            UIInput inputComponent = (UIInput) component;
            inputComponent.setValid(false);
        }
    }

    public static void invalidateInput(FacesMessage facesMessage, String relativeId) {
        FacesContext facesContext = FacesContext.getCurrentInstance();
        UIComponent currentComponent = UIComponent.getCurrentComponent(facesContext);
        if (null == currentComponent) {
            LOGGER.debug("no current component, using view root");
            currentComponent = facesContext.getViewRoot();
        }
        UIComponent component = currentComponent.findComponent(relativeId);
        if (null == component) {
            LOGGER.error("component not found: {}", relativeId);
            facesContext.addMessage(null, facesMessage);
            return;
        }
        String clientId = component.getClientId();
        facesContext.addMessage(clientId, facesMessage);
        if (!(component instanceof UIInput)) {
            LOGGER.error("component is not a UIInput: {}", component.getId());
            return;
        }
        UIInput inputComponent = (UIInput) component;
        inputComponent.setValid(false);
    }
```
Despite having validators, sometimes an action, invoking some business method, will trigger exceptions which you want to blame on a certain input component. The above code makes it easier to mark input components as such.

I've had a look around within OmniFaces and couldn't directly find something like this. Is there something like this within OmniFaces? If not, would be handy...

**Comment by fcorneli** [comment]
Best-effort search within all naming containers:
https://github.com/e-Contract/enterprise-jsf/blob/master/ejsf-taglib/src/main/java/be/e_contract/ejsf/Utils.java
This also allows you to find input components within composites within the form.

**Comment by BalusC** [comment]
I'd like to understand the real world use case for this. 

I guess that [`ValidateMultipleFields`](https://omnifaces.org/docs/javadoc/4.1/org/omnifaces/component/validator/ValidateMultipleFields.html) / `<o:validateXxx>` come most close to what you actually need.

**Comment by BalusC** [comment]
Nevermind, point already taken. I've added it to the `Components` utility class, see above commit. I'm only not torn on taking a whole `FacesMessage` as argument as it tight-couples with responsiblity of `Messages` utility class, so I've replaced it by just a message string + params. 

I've also rewritten it so that the thing also works within UIRepeat/UIData. So you can specify e.g. `1:foo`, `1:bar` etc if you wanted to reference those in the 2nd row.

Let me know if that works out for you.

**Comment by BalusC** [comment]
Already released with 4.2, closing off.
